modules/bullet/btRayShape.cpp

```diff
@@ -97,8 +97,8 @@ void btRayShape::getPreferredPenetrationDirection(int index, btVector3 &penetrat
 
 void btRayShape::reload_cache() {
 
-	m_cacheScaledLength = m_length * m_localScaling[2] + m_collisionMargin;
+	m_cacheScaledLength = m_length * m_localScaling[2];
```

Comment:
Please don't forget to revert this one.

I need that you put this:
`m_cacheScaledLength = m_length * m_localScaling[2] + m_collisionMargin;`

Reply:
If you look at the code more carefully, I moved things a bit around to not have to apply the collision margin in`m_cacheScaledLength`, as that was incorrect.

Why? Because `localGetSupportingVertexWithoutMargin()` would otherwise return the supporting vertex *with* margin and `localGetSupportingVertex()` would return the supporting vertex with the margin applied *twice*.

So AFAICT the margin is applied in all the right places, except maybe `getScaledLength()` which now does not contain the margin. Should it? If yes, why?

Reply:
You true